walrus-tests/tests/round_trip/unreachable_2.wat

```diff
@@ -10,5 +8,6 @@
 ;; CHECK: (module
 ;; NEXT:    (type (;0;) (func (result i32)))
 ;; NEXT:    (func (;0;) (type 0) (result i32)
-;; NEXT:      unreachable)
+;; NEXT:      unreachable
+;; NEXT:      i32.const 42)
```

Comment:
And the idea is we want to write a pass specifically for trimming unreachable isntructions instead of mixing it into `emit`, right?

Reply:
Indeed!